src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -92,6 +92,129 @@ def _get_symbol_table(node):
+        assert isinstance(bound, BinaryOperation), \
+            "Expecting BinaryOperation but found '{0}'." \
+            "".format(type(bound).__name__)
+        assert bound.operator == operator, \
+            "Expecting operator to be '{0}', but found '{1}'." \
+            "".format(str(operator), str(bound.operator))
+        reference = bound.children[0]
+        literal = bound.children[1]
+        assert isinstance(reference, Reference), \
+            "Expecting Reference but found '{0}'." \
+            "".format(type(reference).__name__)
+        assert reference.symbol is array.symbol, \
+            "Expecting Reference symbol '{0}' to be the same as array " \
+            "symbol '{1}'.".format(reference.symbol.name, array.symbol.name)
+        assert isinstance(literal, Literal), \
+            "Expecting Literal but found '{0}'." \
+            "".format(type(literal).__name__)
+        assert literal.datatype == DataType.INTEGER, \
+            "Expecting integer datatype but found '{0}'." \
+            "".format(literal.datatype)
+        assert literal.value == str(dim), \
+            "Expecting literal value '{0}' to be the same as the current " \
+            "array dimension '{1}'.".format(literal.value, str(dim))
+    except AssertionError as excinfo:
+        raise InternalError("psyir/frontend/fparser2.py:check_bound():"
+                            + str(excinfo))
+
+
+def check_literal(node, dim, index, value):
+    '''Utility function to check that the supplied array has a literal at
+    dimension index "dim" and range index "index" with value
+    "value".
+
+    The step part of the range node has an integer literal with
+    value 1 by default.
+
+    Assumes that the node argument is an array and that the
+    supplied dimension index is a Range node and that the supplied
+    range index is valid.
+
+    :param array: the node to check.
+    :type array: :py:class:`pysclone.psyir.node.array`
+    :param int dim: the dimension index to check.
+    :param int index: the index of the range to check (0 is the \
+        lower bound, 1 is the upper bound).
+    :param int value: the expected value of the literal.
+
+    :raises InternalError: if the supplied argument does not have the \
+    required properties.
+
+    '''
+    try:
+        literal = node.children[dim-1].children[index]
+        assert isinstance(literal, Literal), \
+            "Expecting Literal but found '{0}'" \
+            "".format(type(literal).__name__)
+        assert literal.datatype == DataType.INTEGER, \
+            "Expecting integer datatype but found '{0}'." \
+            "".format(literal.datatype)
+        assert literal.value == str(value), \
+            "Expecting literal value '{0}' to be the same as '{1}'." \
+            "".format(literal.value, str(value))
+    except AssertionError as excinfo:
+        raise InternalError("psyir/frontend/fparser2.py:check_literal():"
```

Comment:
I don't think this should be an InternalError - how about a `NotImplementedError`?

Reply:
Changed, with misgivings.

Reply:
Also changed the check_bound function for consistency.